time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java

```diff
@@ -630,4 +615,60 @@ static List<TimeSeries> parseJson(Reader reader) {
     static List<TimeSeries> parseJson(Path file) {
         return JsonUtil.parseJson(file, TimeSeries::parseJson);
     }
+
+    static Instant parseMicrosToInstant(String token) {
+        return parsePreciseDateToInstant(token, 6);
+    }
+
+    static Instant parseNanosToInstant(String token) {
+        return parsePreciseDateToInstant(token, 9);
+    }
+
+    static Instant parsePreciseDateToInstant(String timestamp, int precision) {
+        return timestamp.length() > precision ? Instant.ofEpochSecond(Long.parseLong(timestamp.substring(0, timestamp.length() - precision)),
+            Long.parseLong(timestamp.substring(timestamp.length() - precision) + "0".repeat(9 - precision))) :
+            Instant.ofEpochSecond(0,
+                Long.parseLong(timestamp + "0".repeat(9 - precision)));
+    }
+
+    static String writeInstantToMicroString(Instant instant) {
+        return writeInstantToString(instant, 6);
+    }
+
+    static String writeInstantToNanoString(Instant instant) {
+        return writeInstantToString(instant, 9);
+    }
+
+    static String writeInstantToString(Instant instant, int precision) {
+        long seconds = instant.getEpochSecond();
+        int nanos = instant.getNano();
+        // Compute the multiplier to bring nanos to the right precision
+        long multiplier = (long) Math.pow(10, 9. - precision);
+
+        // Compute the fractional part before rounding off
+        long fraction = nanos / multiplier;
+
+        // Rounding: if the remainder is at least half the multiplier, round up.
+        if (nanos % multiplier * 2 >= multiplier) {
+            fraction++;
+        }
+
+        // If rounding reduces the fraction to 10^(precision), the seconds must be carried forward.
+        long scale = (long) Math.pow(10, precision);
+        if (fraction >= scale && seconds > 0) {
```

Comment:
Can you add a comment to explain the presence of `&& seconds > 0`?
If I am not mistaken, when `fraction >= scale` and `seconds == 0`, both ways lead to the same result, but having the second part of the condition makes that the "shortest" way is taken. Am I right?